recipes/ckanweb-deploy-exts.rb

```diff
@@ -147,15 +147,29 @@
 sorted_plugin_names.concat(dependent_plugin_names)
 sorted_plugin_names.each do |plugin|
 
-	egg_name = `aws ssm get-parameter --region "#{node['datashades']['region']}" --name "/config/CKAN/#{node['datashades']['version']}/app/#{node['datashades']['app_id']}/plugin_apps/#{plugin}/shortname" --query "Parameter.Value" --output text`.strip
+	retries = 0
+	while retries < 5
+		egg_name = `aws ssm get-parameter --region "#{node['datashades']['region']}" --name "/config/CKAN/#{node['datashades']['version']}/app/#{node['datashades']['app_id']}/plugin_apps/#{plugin}/shortname" --query "Parameter.Value" --output text`.strip
```

Comment:
can't we do a single ssm call and then pull apart the parts to their variables instead of lots of little calls.
i.e.

``aws ssm get-parameter --region "#{node['datashades']['region']}" --name "/config/CKAN/#{node['datashades']['version']}/app/#{node['datashades']['app_id']}/plugin_apps/#{plugin}/ ``

this will have ``shortname``, ``type``, ``url``, ``revision``  which we can then pull part.

Reply:
Theoretically, yes. It would be messier since we would need to import the JSON module, use it to parse the `get_parameters` result into a list, then iterate through that list checking "if Name is 'foo' then assign Value to foo, else if name is 'baz' then assign Value to baz, etc". But it could be done.

Reply:
you could try with ruby/json and decrypt the payload. or you could do with get-parameters --names "name1" "name2" "name3"  --query "Parameters[*].[Name,Value]"
and split on name.

```
require 'json'

# Fetch all parameters in a single call
params = JSON.parse(`aws ssm get-parameters-by-path --region "#{node['datashades']['region']}" --path "/config/CKAN/#{node['datashades']['version']}/app/#{node['datashades']['app_id']}/plugin_apps/#{plugin}/app_source/" --query "Parameters[*].[Name,Value]" --output json`.strip)

# Convert the parameters into a hash for easier access
params_hash = params.each_with_object({}) do |(name, value), hash|
  hash[name.split('/').last] = value
end

type = params_hash['type']
revision = params_hash['revision']
url = params_hash['url']
```